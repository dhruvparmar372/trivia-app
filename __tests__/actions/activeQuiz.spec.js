import {
  startQuiz as startQuizAction,
  endQuiz as endQuizAction,
  recordAnswer as recordAnswerAction
} from "source/actions/activeQuiz";

const quizStartDate = "2017-12-30T18:30:00.000Z";
const quizEndDate = "2017-12-31T18:30:00.000Z";
const dummyQuiz = {
  id: 1,
  questions: [
    {
      id: 1,
      type: "boolean",
      question: "1 + 1 = 2?",
      correct_answer: "True",
      incorrect_answers: ["False"]
    }
  ]
};

describe("activeQuiz actions", () => {
  test("start quiz action", () => {
    expect(startQuizAction(dummyQuiz, quizStartDate)).toMatchSnapshot();
  });

  test("end quiz action", () => {
    expect(endQuizAction(dummyQuiz, quizEndDate)).toMatchSnapshot();
  });

  test("end quiz action", () => {
    expect(
      recordAnswerAction(dummyQuiz.id, dummyQuiz.questions[0].id, {
        value: "True"
      })
    ).toMatchSnapshot();
  });
});
