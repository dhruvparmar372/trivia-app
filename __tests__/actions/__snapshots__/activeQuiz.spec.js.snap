// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activeQuiz actions end quiz action 1`] = `
Object {
  "endedOn": "2017-12-31T18:30:00.000Z",
  "quiz": Object {
    "id": 1,
    "questions": Array [
      Object {
        "correct_answer": "True",
        "id": 1,
        "incorrect_answers": Array [
          "False",
        ],
        "question": "1 + 1 = 2?",
        "type": "boolean",
      },
    ],
  },
  "type": "QUIZ/END",
}
`;

exports[`activeQuiz actions end quiz action 2`] = `
Object {
  "answer": Object {
    "value": "True",
  },
  "questionId": 1,
  "quizId": 1,
  "type": "QUIZ/RECORD_ANSWER",
}
`;

exports[`activeQuiz actions start quiz action 1`] = `
Object {
  "quiz": Object {
    "id": 1,
    "questions": Array [
      Object {
        "correct_answer": "True",
        "id": 1,
        "incorrect_answers": Array [
          "False",
        ],
        "question": "1 + 1 = 2?",
        "type": "boolean",
      },
    ],
  },
  "startedOn": "2017-12-30T18:30:00.000Z",
  "type": "QUIZ/START",
}
`;
